### PR TITLE
Updated SMSZB-120 & new converter for genBinaryInput

### PR DESCRIPTION
**Pull request by TheStigh**

### CODE REVIEW

File: converters/fromZigbee.js [hunk 1]
```diff
@@ -6435,6 +6435,28 @@ const converters = {
...
+            }
+            if (msg.data.hasOwnProperty('reliability')) {
+                const lookup = {0: 'no_fault_detected', 7: 'unreliable_other', 8: 'process_error'};
+                result.reliability = lookup[msg.data['reliability']];
+            }
+            if (msg.data.hasOwnProperty('statusFlags')) {
+                const statusFlags = (msg.data.statusFlags & 1<<2) > 0;
+                result.fault = statusFlags;
+            }
+            return result;
```
Comment (Koenkk) [inline converters/fromZigbee.js:6457]: Can you also add these to `exposes`?

File: converters/fromZigbee.js [hunk 2]
```diff
@@ -6435,6 +6435,28 @@ const converters = {
...
+    develco_genbin: {
+        cluster: 'genBinaryInput',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            const result = {};
+            if (msg.data.hasOwnProperty('outOfService')) {
+                result.out_of_service = (msg.data['outOfService']===1);
+            }
+            if (msg.data.hasOwnProperty('presentValue')) {
+                result.present_value = (msg.data['presentValue']===1);
```
Comment (Koenkk) [inline converters/fromZigbee.js:6447]: This name is a bit abstract, what does it mean?

**Comment by rindlerblabla** [comment]
Will you change the voltage converter for the smoke detector as well?

File: converters/fromZigbee.js (hunk 1 above)
Comment (TheStigh) [reply converters/fromZigbee.js:6451]: @Koenkk Done!

File: converters/fromZigbee.js (hunk 2 above)
Comment (TheStigh) [reply converters/fromZigbee.js]: @Koenkk Removed `outOfService` & `presentValue`

**Comment by TheStigh** [comment]
@rindlerblabla Done, both for Smoke, Fire and Siren

**Comment by rindlerblabla** [comment]
Fixes https://github.com/Koenkk/zigbee2mqtt/issues/8457

File: devices/develco.js [hunk 3]
```diff
@@ -195,7 +196,10 @@ module.exports = [
         },
         exposes: [e.temperature(), e.battery(), e.smoke(), e.battery_low(), e.test(),
             exposes.numeric('max_duration', ea.ALL).withUnit('s').withValueMin(0).withValueMax(600).withDescription('Duration of Siren'),
-            exposes.binary('alarm', ea.SET, 'START', 'OFF').withDescription('Manual Start of Siren')],
+            exposes.binary('alarm', ea.SET, 'START', 'OFF').withDescription('Manual Start of Siren'),
```
Comment (Koenkk) [inline devices/develco.js:199]: I think these exposes should be added to SMSZB-120 instead (since you added the fz.develco_genbinaryinput there)

Comment (TheStigh) [reply devices/develco.js:202]: @Koenkk Ah, forgot - thanks for reminding me. Also updated the HESZB with fz.genbinaryinput and genbinaryinput to bind.

File: devices/datek.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ module.exports = [
         model: 'HLU2909K',
         vendor: 'Datek',
         description: 'APEX smart plug 16A',
-        fromZigbee: [fz.on_off, fz.electrical_measurement, fz.metering, fz.temperature],
```
Comment (Koenkk) [inline devices/datek.js:17]: Just to be sure, was this removed on purpose?

Comment (TheStigh) [reply devices/datek.js:17]: Yes, it doesn't support seMetering so no need to keep it there.

**Comment by TheStigh** [comment]
@Koenkk  Became a lot in this PR, but this should be final (I did notice you fixed correct the last commit, thank you)

File: devices/datek.js (hunk 4 above)
Comment (TheStigh) [reply devices/datek.js:17]: es, it doesn't support seMetering so no need to keep it there.

**Comment by Koenkk** [comment]
Thanks again @TheStigh !

**Pull request merged**